packages/cli/lib/lib/webpack/webpack-client-config.js

```diff
@@ -327,6 +316,7 @@ function isDev(env) {
 		},
 		plugins: [
 			new webpack.NamedModulesPlugin(),
+			env.esm && new webpack.HotModuleReplacementPlugin(),
```

Comment:
This is a weird one that I don't fully understand. `webpack-dev-server` v4 will immediately throw with the following error if the `--esm` flag is used:

```
Uncaught Error: [HMR] Hot Module Replacement is disabled.
```

Just points to Webpack itself throwing, so no useful stack trace there. You can see this on the latest release of `preact-cli`.

Users will get warnings that this plugin is automatically added for them if the devServer has `hot` enabled ([as we do](https://github.com/preactjs/preact-cli/blob/a60f8df8e87935f3b3e03a2bb4aecda801568c83/packages/cli/lib/lib/webpack/webpack-client-config.js#L333-L334)), though for whatever reason this does fix the ESM issue. To reduce noise I've therefore put this behind the esm flag check.

Only found it from looking at Prefresh's docs to see if we were missing something. Not sure why it works, but I imagine it's something with `babel-plugin-esm`.